### ISSUE TITLE
braille lags in Windows cmd.exe, worse when using SSH

### ISSUE THREAD

**Issue by mehgcap** [nvaccess/nvda#10600]
### Steps to reproduce:
1. Run cmd.exe. I notice a much longer lag when using SSH to connect to a remote server.

### Actual behavior:
NVDA speech echoes what I type immediately. Without SSH, braille responds in half a second or so. When SSH-ed to a server, it can take up to 2 seconds to respond, This includes typing and deleting text.

### Expected behavior:
Braille and speech should be in sync, as they are in most other areas.

#### NVDA installed/portable/running from source:
I pressed enter on the downloaded file, and chose "continue" instead of choosing to install or make a portable copy.
#### NVDA version:
2019.3 beta 1
#### Windows version:
Windows 10, v1903
#### Name and version of other software in use when reproducing the issue:
cmd.exe built into Windows with SSH Windows feature enabled
#### Does the issue still occur after restarting your PC?
not tested
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2019.2.1 lacks this problem. I've not seen it before in the two years I've been using SSH with Windows and NVDA.

**Comment by lukaszgo1** [comment]
@mehgcap What happens if you set Windows Console support from advanced settings to legacy instead of Prefer UIA?

**Comment by mehgcap** [comment]
With Legacy selected, things seem to work perfectly once more.

**Comment by lukaszgo1** [comment]
cc @codeofdusk @michaelDCurran

**Comment by codeofdusk** [comment]
What happens if you go back to automatic, but select "speak passwords in Windows console"?

**Comment by mehgcap** [comment]
Doing this causes the problem to return. I should also say that the NVDA key sometimes seems to be stuck, and I have to press it (caps lock, in my case) once before it un-sticks. This problem is present regardless of the "speak passwords" option's status. Using legacy mode fixes it, as well as the lag problem. Somehow, the two seem related.

**Comment by codeofdusk** [comment]
> Doing this causes the problem to return. I should also say that the NVDA key sometimes seems to be stuck, and I have to press it (caps lock, in my case) once before it un-sticks. This problem is present regardless of the "speak passwords" option's status. Using legacy mode fixes it, as well as the lag problem. Somehow, the two seem related.

Are you typing on a Braille or qwerty keyboard?

**Comment by mehgcap** [comment]
Sorry, I forgot to specify that. I'm using a standard QWERTY keyboard, using laptop layout, though I'm on a desktop machine. Caps lock is my primary NVDA key, with the insert key in the "six pack" as my alternative.

**Comment by codeofdusk** [comment]
> Sorry, I forgot to specify that. I'm using a standard QWERTY keyboard, using laptop layout, though I'm on a desktop machine. Caps lock is my primary NVDA key, with the insert key in the "six pack" as my alternative.

Do you also experience typing lag in the console when using your Braille keyboard?

**Comment by mehgcap** [comment]
Yes, as well as dropped characters. With legacy mode, I have no trouble. With automatic mode, braille typing is very laggy. I typed
git st
(my alias for "git status) and wound up with
gitt
(note the two Ts). I tried again, and the second word never even appeared. I also got no speech feedback as I deleted using the braille keyboard, except for when I deleted a space. I have no idea what's going on.

**Comment by feerrenrut** [comment]
@mehgcap  Could you please confirm this is resolved with the latest beta?

**Comment by feerrenrut** [comment]
Since we have had no reply from @mehgcap and we believe this is fixed I am going to close the issue.

**Comment by codeofdusk** [comment]
@mehgcap Could you please test if the issue is reproducible under the following conditions? I'd like to re-enable the new console support under the 2021 version of Windows 10 as Microsoft has made a lot of improvements.

1. Run [the latest alpha snapshot](http://nvda-project.org/snapshots).
2. Download OpenConsole [here](https://github.com/nvaccess/nvda/files/4432945/CascadiaPackage_4f8acb4_x64.zip).
3. Enable "use UI automation in Windows Console" from NVDA advanced settings.
4. From the archive, run `OpenConsole.exe`.
5. Test SSH in Braille.

Thanks in advance!

**Comment by mehgcap** [comment]
Okay, @codeofdusk and I just did some testing of this. We used the 21H1 build of conhost, tested in contrast with the default cmd.exe on Windows 10 2004. There's still a slight lag in braille, but it's not bad. Neither terminal we tested was worse than the other. You should be okay to move forward with the plan to use the 21H1 build, at least in regards to this bug.
